SalesforceCommerceCloud/pwa-kit
# Send overridesDir file data to Cloud

# Description

GUS: [W-13505160](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001SZwYOYA1/view)
[Related PR in `portal_app`](https://git.soma.salesforce.com/cc-mobify/portal_app/pull/4546)



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Implemented `walkDir` function that reads then returns a set of file paths that exists within a certain directory
- Invoked the `walkDir` function on both `node_modules/<ccExtensibility.extends>` and `<ccExtensibility.overridesDir>` directories
- Return the intersection of sets as array and set `cc_overrides` field of `bundle_metadata`

# How to Test-Drive This PR

- Generate a PWA Kit project using template extensibility
- Upload the bundle to MRT
- Navigate to [http://cloud-dev.mobify.com:8000/pbjt/core/bundle](http://cloud-dev.mobify.com:8000/pbjt/core/bundle) and click into the bundle detail page of the bundle just uploaded
- Verify under **Bundle Metadata** section that the `cc_overrides` field is populated with the "valid" file paths that you have overridden (valid == files that also exist in the base template)

<img width="1433" alt="Screenshot 2023-08-02 at 3 03 57 PM" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/84923642/84017c0c-4ab4-42e6-9dc3-d59da4d0e32a">

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
